File: src/dataset/javascript/bracket-push_exercism-ThomasZumsteg/claude/ClaudeSonnet4_bracket-push_v2.js
var bracket_pairs = {"{": "}", "[":"]", "(":")"};

function bracket(brackets) {
	/* checks that bracket syntax is correct */
  var stack = [];
  var length = brackets.length;
  
  for(var i = 0; i < length; i++) {
    var letter = brackets[i];
    var closing = bracket_pairs[letter];
    
    if(closing) {
      stack.push(closing);
    } else if(stack.length > 0 && stack[stack.length - 1] === letter) {
      stack.pop();
    } else {
      return false;
    }
  }
  return stack.length === 0;
};

module.exports = bracket;

File: src/dataset/python/bracket-push_exercism-ThomasZumsteg/claude/ClaudeSonnet4_bracket_push_v2.py
MATCHING_BRACKETS = {"(": ")", "{": "}", "[": "]"}
CLOSING_BRACKETS = {")", "}", "]"}

def check_brackets(line):
    """check_brackets checks if brackets are properly balanced"""
    stack = []
    for char in line:
        if char in MATCHING_BRACKETS:
            stack.append(MATCHING_BRACKETS[char])
        elif char in CLOSING_BRACKETS:
            if not stack or char != stack.pop():
                return False
    return not stack